server/backend/background/background.go

```diff
@@ -60,7 +61,11 @@ func New() *Background {
 
 // AttachGoroutine creates a goroutine on a given function and tracks it using
 // the background's WaitGroup.
-func (b *Background) AttachGoroutine(f func(ctx context.Context)) {
+func (b *Background) AttachGoroutine(
+	f func(ctx context.Context),
+	metrics *prometheus.Metrics,
```

Comment:
I think `metrics` parameter will not change every time we call `AttachGoroutine()`. How about putting `metrics` parameter to `Background` struct?

As you can see in the other use cases of metrics, we pull out `metrics` from struct, just like `Backend`.

Reply:
Great idea!
The arguments that go into the parameters are easy things that change, but metrics probably won't. I will be making the changes you suggested.